File: EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/SchoolEventCalendar.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class SchoolEventCalendar : BaseAuditableEntity
{
    public long SchoolId { get; set; }
    public string EventTitleAr { get; set; } = string.Empty;
    public string EventTitleEn { get; set; } = string.Empty;
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int EventType { get; set; } // 1=Holiday, 2=ExamPeriod, 3=Conference, 4=Activity
    public bool IsPublic { get; set; } = true;
    public string? Description { get; set; }
    public long? OrganizerEmployeeId { get; set; }
    public int TargetAudience { get; set; } = 1; // 1=All, 2=Students, 3=Guardians, 4=Staff
    public string? LocationDetails { get; set; }
    public bool RequiresAttendanceTracking { get; set; }

    // Navigation Property
    public virtual School? School { get; set; }
}

File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m1-school-office/interface/school-event-calendar.ts
import type { EventType, TargetAudience } from './common';

export interface SchoolEventCalendar {
  id: number;
  schoolId: number;
  eventTitleAr: string;
  eventTitleEn: string;
  startDate: string;
  endDate: string;
  eventType: EventType;
  isPublic: boolean;
  description: string | null;
  organizerEmployeeId: number | null;
  targetAudience: TargetAudience;
  locationDetails: string | null;
  requiresAttendanceTracking: boolean;
}

export type CreateSchoolEventCalendarDto = Omit<SchoolEventCalendar, 'id'>;

export type UpdateSchoolEventCalendarDto = Omit<SchoolEventCalendar, 'schoolId'>;
